client/src/components/auth/Register.js

```diff
@@ -1,83 +1,93 @@
 import React, { Component } from 'react'
 import PropTypes from 'prop-types';
-import './Register.css'
 import classnames from 'classnames';
 import { connect } from 'react-redux';
 import { registeruser } from '../../actions/authActions';
 
 class Register extends Component {
   constructor() {
     super();
-   this.state = {
-     name: '',
-     email: '',
-     password: '',
-     password2: '',
-     errors: {}
-   } 
-   this.onChange = this.onChange.bind(this);
-   this.onSubmit = this.onSubmit.bind(this);
+    this.state = {
+      name: '',
+      email: '',
+      password: '',
+      password2: '',
+      errors: {}
+    }
+    this.onChange = this.onChange.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
   }
-onChange(e){
-  this.setState({[e.target.name]: e.target.value})
-}
-onSubmit(e){
-  e.preventDefault();
 
-  const newUser = {
-    name: this.state.name,
-    email: this.state.email,
-    password: this.state.password,
-    password2: this.state.password2
-  };
+  onChange(e){
+    this.setState({[e.target.name]: e.target.value});
+  }
 
-  this.props.registeruser(newUser);
```

Comment:
I don't think this is needed as you already state   this.props.registerUser(newUser, this.props.history);
  } on line 35. I've checked on DevConnector on Kal's code and she also just has this once (line 35 that you have) and not here in line 34